### PR TITLE
Improvements to the UI for importing and enriching data [MAP-347] [MAP-78] [MAP-77]

### PR DESCRIPTION
**Pull request by Moggach**
## Description
This PR adds:
if an import or update is in a queue behind others indicate this in the UI
a progress bar when importing and updating records
when an job is going to take longer than 5 minutes tell the user it is finished via email

## Motivation and Context
Addresses issues [MAP-347,](https://linear.app/commonknowledge/issue/MAP-347/when-an-job-is-going-to-take-longer-than-it-is-reasonable-for-someone), [MAP-77](https://linear.app/commonknowledge/issue/MAP-77/no-progress-bar-or-estimated-time-on-imported-records), [MAP-78](https://linear.app/commonknowledge/issue/MAP-78/if-an-import-or-update-is-in-a-queue-behind-others-indicate-this-in)

## How Can It Be Tested?
Run the branch locally
Import and/or update a data source
**Testing progress bar:**

- If working with smaller data sets update the IMPORT_UPDATE_ALL_BATCH_SIZE to a small size e.g. 2
- Observe a progress bar during the import or update

**Testing job queue:**

- Try requesting more than one import/update job at once and observe that the UI indicates that you are in a queue

**Testing email notification:**

- set timedelta in get_scheduled_batch_job_progress to something low (e.g. 1 minute)
- observe that the UI indicates you will be sent an email
- observe the email printed to the console

### CODE REVIEW

File: hub/graphql/types/model_types.py [hunk 1]
```diff
@@ -880,7 +894,7 @@ def is_update_scheduled(self: models.ExternalDataSource, info: Info) -> bool:
     def update_progress(
         self: models.ExternalDataSource, info: Info
     ) -> Optional[BatchJobProgress]:
-        job = self.get_scheduled_update_job()
+        job = self.get_latest_update_job()
         if job is None:
             return None
         progress = self.get_scheduled_batch_job_progress(job)
```
Comment (joaquimds) [inline hub/graphql/types/model_types.py:900]: Do you need to pass the `user` in here, too?

**Review by joaquimds — CHANGES_REQUESTED** [review]
One issue: if the estimated job time is > 5 minutes, and the actual job time is < 5 minutes, then we'll tell people we're going to email them, but then we won't. How can we solve this?

**Comment by Moggach** [comment]
> One issue: if the estimated job time is > 5 minutes, and the actual job time is < 5 minutes, then we'll tell people we're going to email them, but then we won't. How can we solve this?

Good question - maybe we should just use estimated job time across the whole thing? I don't think it's a problem if it's done in less time?

File: hub/graphql/types/model_types.py (hunk 1 above)
Comment (Moggach) [reply hub/graphql/types/model_types.py]: I think I've added that now

**Comment by janbaykara** [comment]
> One issue: if the estimated job time is > 5 minutes, and the actual job time is < 5 minutes, then we'll tell people we're going to email them, but then we won't. How can we solve this?

Tell them "if it takes longer than 5 mins, we'll email".

@Moggach does this code email people if webhooks force an import? I think it should only email for manually requested full imports. Presumably this code does the latter?

**Comment by Moggach** [comment]
> > One issue: if the estimated job time is > 5 minutes, and the actual job time is < 5 minutes, then we'll tell people we're going to email them, but then we won't. How can we solve this?
> 
> Tell them "if it takes longer than 5 mins, we'll email".
> 
> @Moggach does this code email people if webhooks force an import? I think it should only email for manually requested full imports. Presumably this code does the latter?

I don't think we want to do this Tell them "if it takes longer than 5 mins, we'll email". because we are telling them it will take longer than 5 minutes so they can navigate away. So the value is from giving them the heads up which is why I think we should use the estimated job time

**Comment by Moggach** [comment]
> @Moggach does this code email people if webhooks force an import? I think it should only email for manually requested full imports. Presumably this code does the latter?

I think it will fire off for webhooks so will look at this

**Review by ev-sc — CHANGES_REQUESTED** [review]
I get this on the data sources page:

image

**Comment by Moggach** [comment]
I've refactored the timing check to use estimated time only as we need to make the decision before the job is finished

**Comment by janbaykara** [comment]
OK. I think early next week @joaquimds or I just need to test this robustly, fix whatever's required and then merge. It might be right as rain already though, looking good @Moggach!

**Pull request merged**